components/session-based-conditional-auth-functions/src/main/java/org/wso2/carbon/identity/extensions/authenticator/conditional/auth/functions/util/SessionValidationUtil.java

```diff
@@ -0,0 +1,157 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+package org.wso2.carbon.identity.extensions.authenticator.conditional.auth.functions.util;
+
+import org.apache.axis2.transport.http.HTTPConstants;
+import org.apache.commons.httpclient.HttpStatus;
+import org.apache.commons.ssl.Base64;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.ContentType;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.wso2.carbon.identity.application.authentication.framework.model.AuthenticatedUser;
+import org.wso2.carbon.identity.core.util.IdentityUtil;
+import org.wso2.carbon.identity.extensions.authenticator.conditional.auth.functions.exception.SessionValidationException;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
+
+/**
+ * Utility methods used in the session validation conditional authentication functions
+ */
+public class SessionValidationUtil {
+
+    /**
+     * Method to retrieve session data from session data source
+     *
+     * @param authenticatedUser AuthenticatedUser object that represent the user
+     * @return JSON array with each element describing active session
+     * @throws IOException                When it fails to read response from the REST call
+     * @throws SessionValidationException when REST response is not in state 200
+     */
+    public static JSONArray getSessionDetails(AuthenticatedUser authenticatedUser) throws
+            IOException, SessionValidationException {
+
+        HttpClientBuilder httpClientBuilder = HttpClientBuilder.create();
+        HttpClient httpClient = httpClientBuilder.build();
+        HttpPost httpPost = createHttpRequest(authenticatedUser);
+        HttpResponse httpResponse;
+        JSONArray responseJsonArray;
+        httpResponse = httpClient.execute(httpPost);
+        if (httpResponse.getStatusLine().getStatusCode() == HttpStatus.SC_OK) {
+            BufferedReader bufferedReader;
+            bufferedReader = new BufferedReader(new InputStreamReader(httpResponse.getEntity()
+                    .getContent(),
+                    StandardCharsets.UTF_8.name()));
+            StringBuilder responseResult = new StringBuilder();
+            String line;
+            while ((line = bufferedReader.readLine()) != null) {
+                responseResult.append(line);
+            }
+            responseJsonArray = new JSONArray(responseResult.toString());
+            bufferedReader.close();
```

Comment:
close the BufferReader in a finally block.

Reply:
try-with-resources method used